File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/DTO/QueryDTO/IMessageCursorQueryResultDTO.cs
namespace WorldFeed.Common.Public.Models.Interfaces.DTO.QueryDTO
{
    using System.Collections.Generic;

    using WorldFeed.Common.Public.Models.Interfaces.DTO.Events;

    public interface IMessageCursorQueryResultDTO : IBaseCursorQueryDTO<IMessageEventDTO>
    {
        IMessageEventDTO[] MessageEvents { get; set; }

        Dictionary<long, IApp> Apps { get; set; }
    }
}

File: WorldFeed/Client/src/app/core/Public/Models/Interfaces/DTO/QueryDTO/IMessageCursorQueryResultDTO.ts
import {IBaseCursorQueryDTO} from "./IBaseCursorQueryDTO";
import {IMessageEventDTO} from "../Events/IMessageEventDTO";
import Dictionary from "../../../../../../c#-objects/TypeScript.NET-Core/packages/Core/source/Collections/Dictionaries/Dictionary";
import {IApp} from "../../IApp";

export interface IMessageCursorQueryResultDTO extends IBaseCursorQueryDTO<IMessageEventDTO> {
  messageEvents: IMessageEventDTO[];
  apps: Dictionary<number, IApp>;
}
